web_scraping.py

```diff
@@ -33,23 +33,21 @@
 ]
 
 
+def rgb_to_decimal(r, g, b):
+    """converting an rgb value to a base10 value"""
+    return (r << 16) + (g << 8) + b
+
 def get_colors(img_path, num_colors=8):
     """
     Extract num_colors from an image file, returned as a list of RGB Dictionaries
     """
     colors = colorgram.extract(img_path, num_colors)
-    # In the paper they only care about the dominant color feature vector, not all 8 colors
-    dom_proportion_and_color = max(
-        [(c.proportion, c.rgb) for c in colors], key=lambda c: c[0]
-    )
-    dom_color = dom_proportion_and_color[1]
-    # each color is encoded in decimal form?
-    red, green, blue = (
-        float(dom_color.r),
-        float(dom_color.g),
-        float(dom_color.b),
-    )
-    return red, green, blue
+    colors = [(rgb_to_decimal(c.rgb.r, c.rgb.g, c.rgb.b)) for c in colors]
+    output_colors = [0] * 8
```

Comment:
I still ended up using the loop, I tried the other approach, had a big then gave up in the interest of time. Feel free to try to implement it if you want to